reframe/utility/lmod_helper.py

```diff
@@ -0,0 +1,67 @@
+import subprocess
+
+class PopenCommunicate():
+    def exec(self, command):
+        proc=subprocess.Popen(command,
+                              shell = True,
+                              stderr=subprocess.STDOUT,
+                              stdout=subprocess.PIPE)
+        return str(proc.communicate()[0]) # None-terminated, removed None
+
+
+def _all_avail_mods(thing_to_load, ml_av_output):
```

Comment:
I think an extended version of this function should go to the Lmod backend. But it does require an extension of the [Modules' System](https://github.com/eth-cscs/reframe/blob/c9e3406c044843f570223f76d313912cc27795cc/reframe/core/modules.py#L80).